Themis/Models/Authentication.swift

```diff
@@ -18,64 +18,37 @@ class Authentication: NSObject {
-    let keychain: Keychain
+    let url: URL
 
-    override init() {
-        keychain = Keychain(service: "feedback2go.auth") // should this be changed as well?
-    }
+    @objc dynamic var authenticated: Bool = false
 
-    func storeTokenInKeychain(token: String) {
-        DispatchQueue.global().async {
-            do {
-                try self.keychain
-                    .accessibility(.whenPasscodeSetThisDeviceOnly, authenticationPolicy: [.biometryAny])
-                    .set(token, key: "token")
-            } catch let error {
-                print(error.localizedDescription)
-            }
-        }
+    init(for url: URL) {
+        self.url = url
+        super.init()
     }
 
-    func getTokenFromKeychain() {
-        DispatchQueue.global().async {
-            do {
-                self.token = try self.keychain
-                    .authenticationPrompt("Authenticate to login to App")
-                    .get("token")
-            } catch let error {
-                print(error.localizedDescription)
-            }
-        }
-    }
-
-    func deleteToken() {
-        do {
-            try keychain.remove("token")
-            self.token = nil
-        } catch let error {
-            print(error.localizedDescription)
-        }
-    }
-
-    /// This Method allows the user to Authenticate. If it doesnt throw an Error the Bearer token will be set
+    /// This Method allows the user to Authenticate. If it doesnt throw an Error the token will be set as a cookie
     /// If the rememberMe flag is set the token will be valid for 30 Days (if not 30 minutes)
     func auth(username: String, password: String, rememberMe: Bool = false) async throws {
         let body = AuthBody(username: username, password: password, rememberMe: rememberMe)
         let request = Request(method: .post, path: "/api/authenticate", body: body)
-        self.token = try await RESTController.shared.sendRequest(request) { try parseAuth(data: $0) }
+        _ = try await RESTController.shared.sendRequest(request) { _ in true }
+        checkAuth()
     }
 
-    private func parseAuth(data: Data) throws -> String {
-        let decoder = JSONDecoder()
-        let decodedData = try decoder.decode([String: String].self, from: data)
-        guard let token = decodedData["id_token"] else {
-            throw AuthenticationError.tokenNotFound
+    func logOut() async throws {
+        let request = Request(method: .post, path: "/api/logout")
+        _ = try await RESTController.shared.sendRequest(request) { _ in true }
+        checkAuth()
+    }
+
+    func checkAuth() {
+        let cookie = HTTPCookieStorage.shared.cookies(for: url)?.first { cookie in
+            guard let url = url.host(), let date = cookie.expiresDate else {
+                return false
+            }
+            return cookie.name == "jwt" && cookie.domain == url && Date() < date
         }
-        return token
+        self.authenticated = cookie != nil
     }
```

Comment:
The URL check is redundant because of `for: url` I believe (?). Also, I would put the cookie-validation logic outside of the `first` search.
```suggestion
    /// An HTTPCookie is valid if it exists (not nil) and it's not expired.
    private func isValid(_ cookie: HTTPCookie?) -> Bool {
        Date() < cookie?.expiresDate ?? Date()
    }

    func checkAuth() {
        let cookie = HTTPCookieStorage.shared.cookies(for: url)?.first { $0.name == "jwt" }
        self.authenticated = isValid(cookie)
    }
```

Reply:
Deleted domain check as it is redundant. But don't think it makes sense to make an extra function call etc. I think it should be fine like it is now.